lib/rubocop/cop/lint/number_convertion.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for usage of #to_i.
+      #
+      # @example
+      #   # bad
+      #   foo.to_i
+      #
+      #   # good
+      #   Integer(foo)
+      class NumberConvertion < Cop
+        MSG = 'Use `Integer()` instead of `#to_i`.'.freeze
+
+        def_node_matcher :to_i?, <<-PATTERN
+          (send (...) :to_i)
```

Comment:
Use `(send _recv :to_i)` instead. `(...)` matches several children, but method name is always 2nd node.